understory_index/src/backends/grid.rs

```diff
@@ -124,26 +123,32 @@ impl<P: Copy + Debug> Backend<f64, P> for GridF64<P> {
         self.entries.clear();
         self.cells.clear();
     }
-    fn query_point<'a>(&'a self, x: f64, y: f64) -> Box<dyn Iterator<Item = usize> + 'a> {
+
+    fn visit_point<F: FnMut(usize)>(&self, x: f64, y: f64, mut f: F) {
         let key = self.key_for(x, y);
-        let mut set = BTreeSet::new();
+        // Dedup without heap using last_seen + epoch.
         if let Some((_, _, slots)) = self.cells.iter().find(|(cx, cy, _)| (*cx, *cy) == key) {
             for &s in slots {
-                set.insert(s);
+                if let Some(Some(a)) = self.entries.get(s) {
+                    // safety: last_seen has same length as entries
+                    if a.contains_point(x, y) {
+                        f(s);
+                    }
+                }
             }
         }
```

Comment:
This may not deduplicate as expected: if `GridF64::insert` is called twice with the same slot and overlapping geometry, then overlapping cells will get the `slot` pushed twice to the slot `Vec`, and this will then yield the (newest) entry twice. If it can be guaranteed slots in any one `cell` are unique (by not inserting the slot into any cells it is already present in), then iterating over slots shouldn't yield duplicates.